src/ApiClient.php

```diff
@@ -0,0 +1,210 @@
+<?php
+
+namespace BookingProtect\InsuranceHub\Client;
+
+use Exception;
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\ClientException;
+use GuzzleHttp\Exception\GuzzleException;
+use GuzzleHttp\Psr7\Request;
+use GuzzleHttp\RequestOptions;
+use JsonMapper;
+use JsonSerializable;
+use Psr\Http\Message\ResponseInterface;
+
+class ApiClient implements IApiClient {
+    private IAuthTokenGenerator $authTokenGenerator;
+    private ApiClientConfiguration $configuration;
+    private IApiClientUrlBuilder $apiClientUrlBuilder;
+    private JsonMapper $jsonMapper;
+
+    public function __construct(ApiClientConfiguration $configuration, IAuthTokenGenerator $authTokenGenerator, IApiClientUrlBuilder $apiClientUrlBuilder) {
+        $this->authTokenGenerator  = $authTokenGenerator;
+        $this->configuration       = $configuration;
+        $this->apiClientUrlBuilder = $apiClientUrlBuilder;
+        $this->jsonMapper          = new JsonMapper();
```

Comment:
Let's change this to be a dependency on the constructor rather than instantiate it here.  This will improve testability and help those using dependency injection.

Reply:
Fully agree